main/src/test/scala/sbt/internal/TestBuild.scala

```diff
@@ -249,54 +253,56 @@ abstract class TestBuild {
     Structure(env, current, data, KeyIndex(keys, projectsMap, confMap), keyMap)
   }
 
-  implicit lazy val mkEnv: Gen[Env] = {
-    implicit val pGen = (uri: URI) =>
-      genProjects(uri)(nonEmptyId, MaxDepsGen, MaxProjectsGen, cGen.arbitrary)
-    envGen(buildGen(uriGen, pGen), tGen.arbitrary)
+  lazy val mkEnv: Gen[Env] = {
+    val pGen = (uri: URI) => genProjects(uri)(nonEmptyId, MaxDepsGen, MaxProjectsGen, cGen)
+    envGen(buildGen(uriGen, pGen), tGen)
   }
 
-  implicit def maskGen(implicit arbBoolean: Arbitrary[Boolean]): Gen[ScopeMask] = {
-    val b = arbBoolean.arbitrary
+  def maskGen: Gen[ScopeMask] = {
+    val b = Gen.boolean
     for (p <- b; c <- b; t <- b; x <- b)
       yield ScopeMask(project = p, config = c, task = t, extra = x)
   }
 
   val kebabIdGen: Gen[String] = for {
     c <- alphaLowerChar
-    cs <- listOfN(MaxIDSize - 2, frequency(MaxIDSize -> alphaNumChar, 1 -> Gen.const('-')))
+    cs <- Gen.list(
+      Gen.frequency(MaxIDSize -> alphaNumChar, List(1 -> Gen.constant('-'))),
+      Range.linear(0, MaxIDSize - 2)
+    )
     end <- alphaNumChar
   } yield (List(c) ++ cs ++ List(end)).mkString
 
-  val uriChar: Gen[Char] = {
-    frequency(9 -> alphaNumChar, 1 -> oneOf("/?-".toSeq))
-  }
+  val optIDGen: Gen[Option[String]] = Gen.choice1(nonEmptyId.map(some.fn), Gen.constant(None))
 
-  val optIDGen: Gen[Option[String]] =
-    Gen.oneOf(nonEmptyId.map(x => Some(x)), Gen.const(None))
+  val pathGen = for {
+    c <- alphaLowerChar
+    cs <- Gen.list(alphaNumChar, Range.linear(6, MaxIDSize))
+  } yield (c :: cs).mkString
 
   val uriGen: Gen[URI] = {
     for {
-      ssp <- nonEmptyId
+      ssp <- pathGen
       frag <- optIDGen
     } yield new URI("file", "///" + ssp + "/", frag.orNull)
   }
 
-  implicit def envGen(implicit bGen: Gen[Build], tasks: Gen[Vector[Taskk]]): Gen[Env] =
-    for (i <- MaxBuildsGen; bs <- containerOfN[Vector, Build](i, bGen); ts <- tasks)
+  def envGen(bGen: Gen[Build], tasks: Gen[Vector[Taskk]]): Gen[Env] =
+    for (bs <- bGen.list(MaxBuildsGen).map(_.toVector); ts <- tasks)
```

Comment:
@eed3si9n Just a trick I sometimes do to generate unique values in a list, I `zipWithIndex` and then prepend/append to the value that needs to be unique with `_$i` or something like that. Means you don't need to worry about lengths/uniqueness. The only downside is that you're not "testing" things that don't always have `_$i` in that value, which might not test something about the name.